### PR TITLE
Create Searchbar

### PR DESCRIPTION
**Pull request by PierreGoe**

### CODE REVIEW

**Review by HeloiseFerla — DISMISSED** [review]
ok

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,26 @@
+import { useState } from 'react';
+
```
Comment (LoicBrassart) [inline src/App.jsx]: ?

File: src/components/Header/index.jsx [hunk 2]
```diff
@@ -16,7 +22,14 @@ export default function Header({ displaySearch }) {
...
       <div className="bottom">
-        {displaySearch ? <SearchBar /> : null}
+        {displaySearch ? (
+          <SearchBar
+            searchBarInputUser={searchBarInputUser}
+            setSearchBarInputUser={setSearchBarInputUser}
+            statsSearchBar={statsSearchBar}
+            setStatsSearchBar={setStatsSearchBar}
+          />
+        ) : null}
```
Comment (LoicBrassart) [inline src/components/Header/index.jsx]: Make it easier to read:
```
displaySearch &&
          <SearchBar
            searchBarInputUser={searchBarInputUser}
            setSearchBarInputUser={setSearchBarInputUser}
            statsSearchBar={statsSearchBar}
            setStatsSearchBar={setStatsSearchBar}
          />
```

File: src/components/Header/index.jsx [hunk 3]
```diff
@@ -25,5 +38,15 @@ export default function Header({ displaySearch }) {
 
 Header.propTypes = {
   displaySearch: PropTypes.bool,
+  searchBarInputUser: PropTypes.string,
+  setSearchBarInputUser: PropTypes.string,
```
Comment (LoicBrassart) [inline src/components/Header/index.jsx]: `function`

File: src/components/Header/index.jsx [hunk 4]
```diff
@@ -25,5 +38,15 @@ export default function Header({ displaySearch }) {
...
+  statsSearchBar: PropTypes.bool,
+  setStatsSearchBar: PropTypes.bool,
```
Comment (LoicBrassart) [inline src/components/Header/index.jsx]: `function`

File: src/components/SearchBar/index.jsx [hunk 5]
```diff
@@ -1,12 +1,31 @@
+import PropTypes from 'prop-types';
 import searchIcon from '../../assets/SearchIcon.png';
 import './style.css';
 
-export default function SearchBar() {
+export default function SearchBar({
+  searchBarInputUser,
+  setSearchBarInputUser,
+
```
Comment (LoicBrassart) [inline src/components/SearchBar/index.jsx]: ?

File: src/components/SearchBar/index.jsx [hunk 6]
```diff
@@ -1,12 +1,31 @@
...
+        onSubmit={(e) => {
+          e.preventDefault();
+          setStatsSearchBar(true);
+        }}
+      >
         <div>
-          <input type="search" id="maRecherche" name="q" />
+          <input
+            type="search"
+            id="maRecherche"
```
Comment (LoicBrassart) [inline src/components/SearchBar/index.jsx]: English plz !

File: src/components/SearchBar/index.jsx [hunk 7]
```diff
@@ -1,12 +1,31 @@
...
+            name="q"
```
Comment (LoicBrassart) [inline src/components/SearchBar/index.jsx]: Bad variable name : mySearch ?

File: src/components/SearchBar/index.jsx [hunk 8]
```diff
@@ -15,3 +34,14 @@ export default function SearchBar() {
     </div>
   );
 }
+
+SearchBar.propTypes = {
+  searchBarInputUser: PropTypes.string,
+  setSearchBarInputUser: PropTypes.string,
```
Comment (LoicBrassart) [inline src/components/SearchBar/index.jsx]: `function`

File: src/components/SearchBar/index.jsx [hunk 9]
```diff
@@ -15,3 +34,14 @@ export default function SearchBar() {
...
+  setStatsSearchBar: PropTypes.bool,
```
Comment (LoicBrassart) [inline src/components/SearchBar/index.jsx]: `function`

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Please fix !

File: src/App.jsx (hunk 1 above)
Comment (Morel-Victor) [reply src/App.jsx]: white line is removed

File: src/components/Header/index.jsx (hunk 2 above)
Comment (Morel-Victor) [reply src/components/Header/index.jsx]: Done !

File: src/components/Header/index.jsx (hunk 3 above)
Comment (Morel-Victor) [reply src/components/Header/index.jsx]: PropsType modify

File: src/components/Header/index.jsx (hunk 4 above)
Comment (Morel-Victor) [reply src/components/Header/index.jsx]: PropsType modify

File: src/components/SearchBar/index.jsx (hunk 5 above)
Comment (Morel-Victor) [reply src/components/SearchBar/index.jsx]: white line are revoded

File: src/components/SearchBar/index.jsx (hunk 6 above)
Comment (Morel-Victor) [reply src/components/SearchBar/index.jsx]: variable name is change

File: src/components/SearchBar/index.jsx (hunk 7 above)
Comment (Morel-Victor) [reply src/components/SearchBar/index.jsx]: variable name is change

File: src/components/SearchBar/index.jsx (hunk 8 above)
Comment (Morel-Victor) [reply src/components/SearchBar/index.jsx]: PropsType modify

File: src/components/SearchBar/index.jsx (hunk 9 above)
Comment (Morel-Victor) [reply src/components/SearchBar/index.jsx]: PropsType modify

**Review by HeloiseFerla — APPROVED** [review]
ok

**Review by LoicBrassart — APPROVED** [review]
No more Evils

**Pull request merged**